src/template/render.rs

```diff
@@ -471,9 +473,45 @@ impl<'a> ChainTemplateSource<'a> {
         };
         debug_span!("Executing command", ?command)
             .in_scope(|| async {
-                let output = Command::new(program)
+                // Spawn the command process
+                let mut process = Command::new(program)
                     .args(args)
-                    .output()
+                    .stdin(Stdio::piped())
+                    .stdout(Stdio::piped())
+                    .stderr(Stdio::piped())
+                    .spawn()
+                    .map_err(|error| ChainError::Command {
+                        command: command.to_owned(),
+                        error,
+                    })
+                    .traced()?;
+
+                // Write to stdin if present
+                if let Some(template) = stdin {
+                    let rendered = template
+                        .render_stitched(context)
+                        .await
+                        .map_err(|error| ChainError::Nested {
+                            field: "stdin".into(),
+                            error: error.into(),
+                        })?;
```

Comment:
We should do this *before* spawning the process. If the command has some sort of side-effect, it would be confusing if the render fails but the side-effect still occurs. This should be outside the tracing span too, since this isn't part of the command execution.